openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/slack/SlackApp.java

```diff
@@ -0,0 +1,287 @@
+package org.openmetadata.service.apps.bundles.slack;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.slack.api.Slack;
+import com.slack.api.bolt.App;
+import com.slack.api.bolt.AppConfig;
+import com.slack.api.methods.SlackApiException;
+import com.slack.api.methods.request.conversations.ConversationsListRequest;
+import com.slack.api.methods.response.conversations.ConversationsListResponse;
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.ws.rs.core.Response;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import org.openmetadata.schema.service.configuration.slackApp.SlackAppConfiguration;
+import org.openmetadata.schema.settings.Settings;
+import org.openmetadata.schema.settings.SettingsType;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.apps.AbstractNativeApplication;
+import org.openmetadata.service.apps.AppException;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.jdbi3.SystemRepository;
+import org.openmetadata.service.search.SearchRepository;
+import org.openmetadata.service.util.JsonUtils;
+
+@Slf4j
+public class SlackApp extends AbstractNativeApplication {
+  private App slackAppInstance;
+  private SlackAppConfiguration appConfig;
+  private SystemRepository systemRepository;
+
+  private static final SecureRandom SECURE_RANDOM = new SecureRandom();
+  private static final Base64.Encoder BASE64_ENCODER = Base64.getUrlEncoder();
+
+  private static final String SLACK_OAUTH_BASE_URL = "https://slack.com/oauth/v2/authorize";
```

Comment:
can we use slack sdk instead of hardcoding urls

Reply:
Noted.

Reply:
Updated.